core/src/main/scala/chisel3/properties/Property.scala

```diff
@@ -335,6 +338,66 @@ private[chisel3] object ClassTypeProvider {
   }
 }
 
+/** Typeclass for Property arithmetic.
+  */
+@implicitNotFound("arithmetic operations are not supported on Property type ${T}")
+trait PropertyArithmeticOps[T] {
```

Comment:
```suggestion
sealed trait PropertyArithmeticOps[T] {
```
This should be sealed right?

Reply:
Oh, yes, I don't see why not.